src/examples/TemplateModule.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useState } from "react";
+import { Form, Input, Grid, Card, Statistic } from "semantic-ui-react";
+
+import TxButton from "../TxButton";
+
+export default function Transfer(props) {
+  const { api, accountPair } = props;
+
+  // The transaction submission status
+  const [status, setStatus] = useState("");
+
+  // The currently stored value
+  const [currentValue, setCurrentValue] = useState(0);
+  const [formValue, setFormValue] = useState(0);
+
+  useEffect(() => {
+    let unsubscribe;
+    api.query.templateModule.something(newValue => {
+      // The storage value is an Option<u32>
+      // So we have to check whether it is None first
+      // There is also unwrapOr
+      if (newValue.isNone){
+        setCurrentValue("<None>");
+      }
+      else{
+        setCurrentValue(newValue.unwrap().toNumber());
+      }
+    }).then(unsub => {
+      unsubscribe = unsub;
+    })
+    .catch(console.error);
+
+    return () => unsubscribe && unsubscribe();
+
+  }, [api.query.templateModule, api.query.templateModule.something]);
```

Comment:
I don't think you need both. We should figure out what precisely you need so that future people don't also think they need both.

Reply:
You're right. Seems like it still works fine with only `api.query.templateModule`.

@Tbaut Can you help us understand what exactly needs to go in that array?

Reply:
Things will work fine with nothing in that array as well. This is a dependency array that needs to contains whatever your code depend on. As soon as one of the dep changes, `useEffect` will be triggered again. In this case `api.query.templateModule.something` is a subset of `api.query.templateModule`. The former changes, the latter will change as well, but the opposite is not true. `api.query.templateModule` is more general so you can keep that one only.